packages/language-core/lib/codegen/script/component.ts

```diff
@@ -52,8 +52,7 @@ export function* generateComponent(
 export function* generateComponentSetupReturns(scriptSetupRanges: ScriptSetupRanges): Generator<Code> {
 	// fill $props
 	if (scriptSetupRanges.props.define) {
-		// NOTE: defineProps is inaccurate for $props
-		yield `$props: __VLS_makeOptional(${scriptSetupRanges.props.name ?? `__VLS_props`}),${newLine}`;
+		yield `$props: ${scriptSetupRanges.props.name ?? `__VLS_props`},${newLine}`;
```

Comment:
Not sure if this is correct in general.

Try with a component with props:

```
  msg: {
    type: String,
    default: null
  }
```

In that case the prop should be optional but won't.

I feel like that might need coordination with changes in Vue types.

Reply:
I just add a test case in https://github.com/vuejs/language-tools/pull/4878/commits/3557d8b17a9e5e78c18013262823649410effada for this.